vault/login_mfa.go

```diff
@@ -2032,26 +2032,25 @@ func (c *Core) validateOkta(ctx context.Context, mConfig *mfa.Config, username s
 	var factorFound bool
 	var userFactor *okta.UserFactor
 	for _, factor := range factors {
-		if factor.IsUserFactorInstance() {
-			userFactor = factor.(*okta.UserFactor)
-			if userFactor.FactorType == "push" {
-				factorFound = true
-				break
-			}
+		f := factor.GetActualInstance()
+		userFactor = f.(*okta.UserFactor)
+		if *userFactor.FactorType == "push" {
+			factorFound = true
+			break
 		}
 	}
 
 	if !factorFound {
 		return fmt.Errorf("no push-type MFA factor found for user")
 	}
 
-	result, _, err := client.UserFactor.VerifyFactor(ctx, user.Id, userFactor.Id, okta.VerifyFactorRequest{}, userFactor, nil)
+	result, _, err := client.UserFactorAPI.VerifyFactor(ctx, user.GetId(), userFactor.GetId()).Execute()
```

Comment:
The difference between the approach here and the approach in `path_login.go` is for 'light touch' reasons - `path_login.go` uses the "shim", since the setting a token is optional there.  The code here already used the okta API directly, and so the updated version can be similarly updated, without using the "write the request yourself" escape hatch.